### PR TITLE
Indenting and translating to spanish json-apis-and-ajax. 

### LINKED ISSUE #4545: Spanish translation of JSON APIs and Ajax
- [x] Trigger Click Events with jQuery
- [x] Change Text with Click Events
- [x] Get JSON with the jQuery getJSON Method
- [x] Convert JSON Data to HTML
- [x] Render Images from Data Sources
- [x] Prefilter JSON
- [x] Get Geo-location Data

### PR DESCRIPTION
**Pull request by vtamara**
Closes #4545

### CODE REVIEW

**Comment by bugron** [comment]
@justinian336 @Oxyrus would you mind to check this translation, please, if you have some time?

**Comment by Oxyrus** [comment]
I can't, my laptop crashed so I don't have a workstation at the moment.

File: seed/challenges/json-apis-and-ajax.json [hunk 1]
```diff
@@ -49,11 +49,22 @@
         "</div>"
       ],
       "challengeType": 0,
-      "type": "waypoint"
+      "type": "waypoint",
+      "nameEs": "Activa eventos de pulsación con jQuery",
+      "descriptionEs": [
+        "En esta sección, vamos a aprender cómo obtener datos de las APIs. Las APIs - o interfaces de programación de aplicaciones - son herramientas que utilizan los computadores para comunicarse entre sí.",
+        "También aprenderemos cómo actualizar HTML con los datos que obtenemos de estas API usando una tecnología llamada Ajax.",
+        "En primer lugar, vamos a revisar lo que hace la función <code>$(document).ready()</code>. Esta función hace que todo el código dentro de ella se ejecute sólo cuando nuestra página ha sido cargada.",
```
Comment (justinian336) [inline seed/challenges/json-apis-and-ajax.json]: `sólo hasta que` en vez de `sólo cuando`, porque la idea es que no corre antes de que la página se haya cargado. No es tan grave, sólo una sugerencia.

File: seed/challenges/json-apis-and-ajax.json [hunk 2]
```diff
@@ -153,11 +171,25 @@
...
       ],
       "challengeType": 0,
-      "type": "waypoint"
+      "type": "waypoint",
+      "nameEs": "Obtén JSON con el método getJSON de jQuery",
+      "descriptionEs": [
+        "También puedes solicitar los datos de una fuente externa. Aquí es donde entran en juego las API. ",
+        "Recuerda que las API - o interfaces de programación de aplicaciones - son herramientas que utilizan los ordenadores para comunicarse entre sí.",
+        "La mayoría de las API para el web transfieren datos en un formato llamado JSON. JSON significa notación de objeto JavaScript (<em>JavaScript Object Notation</em>).",
+        "Tú ya has estado usando JSON cuando creas un objeto en JavaScript. JSON no es más que las propiedades del objeto y sus valores actuales, intercalados entre un <code>{</code> y un <code>}</code>. ",
```
Comment (justinian336) [inline seed/challenges/json-apis-and-ajax.json]: `ya has utilizado JSON cada vez que creaste` en vez de `ya has estado usando JSON cuando creas`

File: seed/challenges/json-apis-and-ajax.json [hunk 3]
```diff
@@ -215,11 +248,27 @@
         "</div>"
       ],
       "challengeType": 0,
-      "type": "waypoint"
+      "type": "waypoint",
+      "nameEs": "Convierte datos JSON a datos HTML",
+      "descriptionEs": [
+        "Ahora que estamos obteniendo datos de una API JSON, mostremoslos en nuestro HTML.",
```
Comment (justinian336) [inline seed/challenges/json-apis-and-ajax.json]: `mostrémoslos` en vez de `mostremoslos`.

File: seed/challenges/json-apis-and-ajax.json [hunk 4]
```diff
@@ -215,11 +248,27 @@
...
+        "Podemos usar el método <code>.forEach()</code> para recorrer nuestros datos y modificar nuestros elementos HTML.",
+        "En primer lugar, vamos a declarar una variable html con <code>var html = \"\";</code>.",
```
Comment (justinian336) [inline seed/challenges/json-apis-and-ajax.json]: `variable llamada html` en lugar de `variable html`. Alguien puede pensar que html es un tipo de variable.

File: seed/challenges/json-apis-and-ajax.json [hunk 5]
```diff
@@ -215,11 +248,27 @@
...
+        "Después, iteremos por nuestro JSON, añadiendo más HTML a esa variable. Cuando se termina el ciclo, vamos a presentarla. ",
```
Comment (justinian336) [inline seed/challenges/json-apis-and-ajax.json]: `iteremos a través` en vez de `iteremos por`.

File: seed/challenges/json-apis-and-ajax.json [hunk 6]
```diff
@@ -279,11 +328,19 @@
         "</div>"
       ],
       "challengeType": 0,
-      "type": "waypoint"
+      "type": "waypoint",
+      "nameEs": "Presentar Imágenes de Fuentes de Datos",
```
Comment (justinian336) [inline seed/challenges/json-apis-and-ajax.json]: `Presentar imágenes de fuentes de datos` en vez de `Presentar Imágenes de Fuentes de Datos`

File: seed/challenges/json-apis-and-ajax.json [hunk 7]
```diff
@@ -279,11 +328,19 @@
...
+      "descriptionEs": [
+        "Tanto en el JSON que recibimos de la API de fotos de gato de Free Code Camp, como en ",
```
Comment (justinian336) [inline seed/challenges/json-apis-and-ajax.json]: Esta parte está rara en la versión en Inglés, sería bueno solicitar que lo arreglen.

File: seed/challenges/json-apis-and-ajax.json [hunk 8]
```diff
@@ -346,11 +403,20 @@
         "</div>"
       ],
       "challengeType": 0,
-      "type": "waypoint"
+      "type": "waypoint",
+      "nameEs": "Pre-filtra JSON",
+      "descriptionEs": [
+        "Si no queremos presentar cada foto de gato que obtengamos de la API JSON de fotos de gato de Free Code Camp, podemos pre-filtrar el json antes de recorrerlo en un ciclo.",
```
Comment (justinian336) [inline seed/challenges/json-apis-and-ajax.json]: `iterar a través de este` en lugar de `recorrerlo en un ciclo`.

File: seed/challenges/json-apis-and-ajax.json [hunk 9]
```diff
@@ -380,7 +446,20 @@
         "</div>"
       ],
       "challengeType": 0,
-      "type": "waypoint"
+      "type": "waypoint",
+      "nameEs": "Recibir datos de Geo-localización",
+      "descriptionEs": [
+        "Otra cosa interesante que podemos hacer es acceder a la ubicación actual de nuestros usuarios. Cada navegador ha incorporado un geo-localizador que nos puede dar esta información. ",
```
Comment (justinian336) [inline seed/challenges/json-apis-and-ajax.json]: `Todos los navegadores incorporan` en vez de `cada navegador ha incorporado`.

File: seed/challenges/json-apis-and-ajax.json [hunk 10]
```diff
@@ -380,7 +446,20 @@
...
+        "El navegador puede obtener la longitud y latitud actual de nuestros usuarios.",
+        "Como usuario verás un mensaje para permitir o bloquear que el sitio conozca tu ubicación actual. El desafío se puede completar de cualquier manera, siempre y cuando el código sea correcto. ",
```
Comment (justinian336) [inline seed/challenges/json-apis-and-ajax.json]: `evitar` en vez de `bloquear`.

File: seed/challenges/json-apis-and-ajax.json [hunk 11]
```diff
@@ -380,7 +446,20 @@
...
+        "Si lo permites, verás que el texto en el teléfono de sálida cambiará con tu latitud y longitud",
```
Comment (justinian336) [inline seed/challenges/json-apis-and-ajax.json]: `de la derecha` en lugar de `de sálida`.

File: seed/challenges/json-apis-and-ajax.json (hunk 1 above)
Comment (vtamara) [inline seed/challenges/json-apis-and-ajax.json]: De acuerdo. Gracias.

File: seed/challenges/json-apis-and-ajax.json (hunk 2 above)
Comment (vtamara) [inline seed/challenges/json-apis-and-ajax.json]: Deje "ya has usado JSON al crear objetos en JavaScript"

File: seed/challenges/json-apis-and-ajax.json (hunk 3 above)
Comment (vtamara) [inline seed/challenges/json-apis-and-ajax.json]: De acuerdo, gracias.

File: seed/challenges/json-apis-and-ajax.json (hunk 4 above)
Comment (vtamara) [inline seed/challenges/json-apis-and-ajax.json]: De acuerdo. Gracias.

File: seed/challenges/json-apis-and-ajax.json (hunk 5 above)
Comment (vtamara) [inline seed/challenges/json-apis-and-ajax.json]: Bueno. Gracias.

File: seed/challenges/json-apis-and-ajax.json (hunk 6 above)
Comment (vtamara) [inline seed/challenges/json-apis-and-ajax.json]: De acuerdo, gracias.

File: seed/challenges/json-apis-and-ajax.json (hunk 8 above)
Comment (vtamara) [inline seed/challenges/json-apis-and-ajax.json]: Bueno, gracias.

File: seed/challenges/json-apis-and-ajax.json (hunk 9 above)
Comment (vtamara) [inline seed/challenges/json-apis-and-ajax.json]: De acuerdo. Gracias.

File: seed/challenges/json-apis-and-ajax.json (hunk 10 above)
Comment (vtamara) [inline seed/challenges/json-apis-and-ajax.json]: Bueno.

File: seed/challenges/json-apis-and-ajax.json (hunk 11 above)
Comment (vtamara) [inline seed/challenges/json-apis-and-ajax.json]: Bueno gracias.

**Comment by vtamara** [comment]
@bugron, I wanted to tell you that @justinian336  already checked this translation and I already applied his suggestions. Thanks for good work.

**Comment by bugron** [comment]
@vtamara OK,will merge soon, then. Thanks all for your hard work.

**Comment by bugron** [comment]
Thanks @vtamara and @justinian336, merged.
